be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -275,61 +304,336 @@ Status JsonReader::close() {
-                        if (strcmp(column_name, "__op") == 0) {
-                            // special treatment for __op column, fill default value '0' rather than null
-                            column->append_strings(literal_0_slice_vector);
-                        } else {
-                            column->append_nulls(1);
-                        }
-                    } else {
-                        _construct_column((*objectValue)[column_name], column.get(), slot_desc->type());
-                    }
-                }
+                RETURN_IF_ERROR(_process_object(chunk, ordered_slot_descs, obj));
             } else {
-                size_t slot_size = slot_descs.size();
-                size_t jsonpath_size = _scanner->_json_paths.size();
-                for (size_t i = 0; i < slot_size; i++) {
-                    if (slot_descs[i] == nullptr) {
-                        continue;
-                    }
-                    ColumnPtr& column = chunk->get_column_by_slot_id(slot_descs[i]->id());
-                    if (i >= jsonpath_size) {
-                        column->append_nulls(1);
-                        continue;
-                    }
-                    rapidjson::Value* json_values = JsonFunctions::get_json_object_from_parsed_json(
-                            _scanner->_json_paths[i], objectValue, _origin_json_doc.GetAllocator());
-                    if (json_values == nullptr) {
-                        column->append_nulls(1);
-                    } else {
-                        _construct_column(*json_values, column.get(), slot_descs[i]->type());
-                    }
-                }
+                RETURN_IF_ERROR(_process_object_with_json_path(chunk, slot_descs, obj));
+            }
+            break;
+        }
+
+        default: {
+            std::string err_msg("JSON data type is not supported");
+            _state->append_error_msg_to_file("", err_msg);
+            _counter->num_rows_filtered++;
+            return Status::DataQualityError(err_msg.c_str());
+        }
+        }
+    }
+    return Status::OK();
+}
+
+// Reorder the slot_descs as the key order in json document.
+void JsonReader::_reorder_column(std::vector<SlotDescriptor*>& slot_descs,
+                                 simdjson::ondemand::document_reference& doc) {
+    simdjson::ondemand::object obj;
+    auto err = doc.get_object().get(obj);
+    if (err) {
+        return;
+    }
+
+    std::map<std::string, SlotDescriptor*> slot_desc_dict;
+
+    for (auto& desc : slot_descs) {
+        slot_desc_dict[desc->col_name()] = desc;
+    }
+
+    std::ostringstream oss;
+    simdjson::ondemand::raw_json_string json_str;
+
+    std::vector<SlotDescriptor*> ordered_slot_descs;
+    ordered_slot_descs.reserve(slot_desc_dict.size());
+
+    for (auto field : obj) {
+        auto err = field.key().get(json_str);
+        if (err) {
+            return;
```

Comment:
Is it ok to just ignore this error?

Reply:
Nothing would be done to the `slot_descs` if got any error, which only brings a performance penalty.